### PR TITLE
Remove redundant parentheses in `case` statement `if` guards

### PR DESCRIPTION
**Pull request by dosisod**
### Description

A follow up to #4024 but for `if` guards in `case` statements. I noticed this when #4024 was made stable, and noticed I had some code that had extra parens around the `if` guard.

There might be some edge cases I am not thinking of, but all tests are passing. I put these new changes under the `remove_redundant_guard_parens` preview name.

### Checklist

- [x] Add an entry in `CHANGES.md` if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?

### CODE REVIEW

File: tests/data/cases/remove_redundant_parens_in_case_guard.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# flags: --minimum-version=3.10 --preview
+
+match 1:
+    case _ if (True):
+        pass
+
+
+match 1:
+    case _ if (
+        True
```
Comment (JelleZijlstra) [inline tests/data/cases/remove_redundant_parens_in_case_guard.py:10]: Add tests with a comment inside the parentheses in various places

**Review by JelleZijlstra — APPROVED** [review]
Thanks!
